### PR TITLE
godb: create random DB name when create new database

### PR DESCRIPTION
**Pull request by EmilGeorgiev**
Here we create random DB name every time when client invoke NewDatabase function. Also we implemnt method for drop database

### CODE REVIEW

File: datastoretest/mongo.go [hunk 1]
```diff
@@ -77,6 +82,14 @@ func (db *DB) Clean() {
 	}
 }
 
+// Drop removes the entire database including all of its collections.
+func (db *DB) Drop() {
```
Comment (mgenov) [inline datastoretest/mongo.go]: Drop could be performed in Clean directly as I don't see how "Drop" will be distinguished from "Clean" in tests. 

db := NewDatabaseWithHost()
defer db.Clean()
defer db.Drop()  ???

File: datastoretest/mongo.go [hunk 2]
```diff
@@ -20,8 +22,10 @@ type DB struct {
 	database *mgo.Database
 }
 
-// NewDatabase is establishing a new database connection using host from the environment.
-// The variable name for the host is TEST_DB_HOST.
+// NewDatabase is establishing a new database connection using host
+// from the environment. The variable name for the host is TEST_DB_HOST.
+// Also this function create random DB name. You should drop database
```
Comment (mgenov) [inline datastoretest/mongo.go]:
```go
// Testing database uses random database name to ensure consistency in tests. 
// The created database will be dropped after Clean/Drop function is called. 
```

File: datastoretest/mongo.go (hunk 1 above)
Comment (EmilGeorgiev) [reply datastoretest/mongo.go]: I imagine our code:
```go
func TestMain(m *testing.M) {
	db = datastoretest.NewDatabase()

	code := m.Run() // this method will return when all tests finish
 
	db.Drop()
	db.Close()
	os.Exit(code)
}
```

and in each test case:

```go
func Test......(t *testtin.T) {
      db.Clean()
      ...
}
```

Comment (EmilGeorgiev) [reply datastoretest/mongo.go]: Client of godb can forget to invoke Drop method. Therefore i will drop database in Client method.

File: datastoretest/mongo.go (hunk 2 above)
Comment (EmilGeorgiev) [reply datastoretest/mongo.go]: Done

**Review by FeNoMeNa — APPROVED** [review]
LGTM

**Comment by mgenov** [comment]
LGTM !

**Pull request merged**